diff --git a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
index 070257f072..61fd0c8f3b 100644
--- a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
+++ b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
@@ -71,10 +71,10 @@ constexpr int CursorMetaSize(int w, int h) {
           w * h * kCursorBpp);
 }
 
-constexpr PipeWireVersion kDmaBufModifierMinVersion = {.major = 0,
+const PipeWireVersion kDmaBufModifierMinVersion = {.major = 0,
                                                        .minor = 3,
                                                        .micro = 33};
-constexpr PipeWireVersion kDropSingleModifierMinVersion = {.major = 0,
+const PipeWireVersion kDropSingleModifierMinVersion = {.major = 0,
                                                            .minor = 3,
                                                            .micro = 40};
 
